server/database.js

```diff
@@ -88,6 +88,13 @@ function getSubmissions(req, res, next) {
 }
 
 function approveAlgorithm(req, res, next) {
+	// Error if no user logged in or user not an admin
+	if (!"user" in req) {
+		return res.status(400).json("Must be logged in");
```

Comment:
is there a fixed vocabulary of machine-readable error codes?  i've found that to be a useful pattern.  it makes client-side code more resilient against small changes in error messages.

Reply:
To clarify: on DSS, we have a structured error object with mandatory fields.  One is a human-readable message ("Must be logged in") and another is a machine-readable message ("NOT_LOGGED_IN").  We could tweak the human-readable message for clarity, but we would avoid changing the machine-readable message.  The idea is that you can build client logic that rely on the machine-readable message.